### DISCUSSION TITLE
Nested write with an aggregation

### DISCUSSION THREAD

**Discussion by supernovic** [prisma/prisma#14007] [Q&A] [answered]
Hey Prisma team + community, this is a problem that I've encountered and haven't quite found a solution to yet- hoping for some assistance! Essentially I have 2 models:

```
model Parent:
   totalBudget: number
   
model Child:
   budget: number
```

The parent model has a one-to-many relationship with the child model. Whenever I create a new child model, I want to update the parent's `totalBudget` field by aggregating all of the children's budgets, including the newly created one. From what I understand, this is a dependent write that _should_ be accomplished via a nested write, something like this:

```
prisma.parent.update({
    data: { 
        totalBudget: await prisma.child.aggregate({ _sum: { budget: true }}),
        child: { create: { ...newChild } }
     }
 })
 ```
 
 But I have not been able to get this to work, probably due to the nature of the aggregation not having access to the new child record. I have also attempted to use the `$transaction` API to create the child first and then recompute the parent's totalBudget, but to no avail- no matter what I've done, I am unable to include the new child's budget into the aggregation. Please let me know if I'm missing something here, would appreciate any help. Thank you!

**Comment by austincrim** [comment, marked as answer]
Hey @supernovic 👋,

My recommendation would have been to do the two operations separately inside a transaction as you attempted. What about that approach did not work? Did you get a specific error message?

**Reply by supernovic** [reply]
Hey @austincrim , thanks for the response! So essentially I had 3 steps in my transaction

1. create child
2. aggregate/sum budget of all children
3. update parent with the sum

The implementation looked a little like this:

```
         await this.prisma?.$transaction(async prisma => {
                const newChild = await prisma.child.create({
                    data: {
                        budget: newChildRequest.budget,
                        parent: { connect: { id: parentId } }
                    }
                });

                const totalBudget = await this.prisma?.child.aggregate({ _sum: { budget: true}, where: { id: parent id}})._sum.budget;

                await prisma.parent.update({
                    where: { id: parent id },
                    data: {
                        totalBudget
                    }
                });
          }
```

What I was noticing was that the aggregation in step 2 would never include the new child's budget- it would only sum up the budgets of the existing children. Is there something that I'm missing here or some additional call that I need to make?

**Reply by supernovic** [reply]
So I played around with my implementation some more and I got it to work! For future reference, my issue was that my implementation of step 2 was actually in a method `sumChildBudgets()`. That method referenced `this.prisma()` as I had the Prisma client stored as an instance variable of the class where this logic was contained. Instead of making the call to `sumChildBudgets()`, I referenced the `prisma` parameter that was passed into the transaction block and everything worked as expected! 

I'm guessing that the `prisma` parameter that is passed into the transaction block has the most up to date connection/data with the DB, or something like that?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
